lib/runtime.js

```diff
@@ -1,5 +1,6 @@
 var Entry = require("./entry.js").Entry;
 var utils = require("./utils.js");
+var dynRequire = module.require || __non_webpack_require__;
 
 exports.enable = function (Module) {
```

Comment:
This is looking good, just one more issue: `module.require` is an unbound function that refers to `this` ([here](https://github.com/nodejs/node/blob/5bda5faffdeefa448965159e524eacdd14490436/lib/module.js#L513)), so I think this needs to be
```js
var dynRequire = module.require ? module.require.bind(module) : require;
```